src/core.js

```diff
@@ -25,6 +31,14 @@ jQuery.fn.init = function( arg1 ) {
 };
 jQuery.fn.init.prototype = jQuery.fn;
 
+jQuery.htmlPrefilter = function( html ) {
+	var changed = html.replace( rxhtmlTag, "<$1></$2>" );
+	if ( changed !== html && makeMarkup( html ) !== makeMarkup( changed ) ) {
+		migrateWarn( "HTML tags must be properly nested and closed: " + html );
+	}
+	return html;
+};
```

Comment:
This will make `jQuery.htmlPrefilter` in jQuery <3.5 to behave like an identity function which we should avoid IMO. I'd keep the warning mechanism as is as it will prepare users for a 3.5 upgrade but instead of returning `html` I'd return the result of applying the original cached `htmlPrefilter`:
```suggestion
var origHtmlPrefilter = jQuery.htmlPrefilter;
jQuery.htmlPrefilter = function( html ) {
	var changed = html.replace( rxhtmlTag, "<$1></$2>" );
	if ( changed !== html && makeMarkup( html ) !== makeMarkup( changed ) ) {
		migrateWarn( "HTML tags must be properly nested and closed: " + html );
	}
	return origHtmlPrefilter( html );
};
```